lib/waterline/utils/helpers.js

```diff
@@ -87,3 +87,29 @@ exports.matchMongoId = function matchMongoId(id) {
   ) return false;
   else return id.toString().match(/^[a-fA-F0-9]{24}$/) ? true : false;
 };
+
+/**
+ * Merge set default criteria.where values
+ * Mutates the criteria.where object
+ *
+ * @object {object} criteria object
+ * @object {object} default where values
+ * @api public
+ */
+exports.mergeCriteria = function mergeCriteria(criteria,defaultCriteria){
```

Comment:
This feels like a good function to build some unit tests for, especially if we put in handling the `or` case